merlin/models/tf/predictions/classification.py

```diff
@@ -63,3 +95,445 @@ def __init__(
             name=name,
             **kwargs,
         )
+
+
+@tf.keras.utils.register_keras_serializable(package="merlin.models")
+class CategoricalPrediction(ContrastivePredictionBlock):
```

Comment:
@sararb I am wondering if we could make the API simpler for the most common use case, where the category to be predicted is an item id.
Do you think it would make sense to create a child class `ItemPrediction` inheriting from `CategoricalPrediction`, where `target_name` is set automatically to be the Tags.ITEM_ID, and with a boolean flag to enable or not weight_tying?

Reply:
The weight-tying is defined by explicitly providing the embedding table of the categorical feature (item-id in the common use-case) because the new implementation of the prediction blocks does not have access to the context layer anymore (We aim to remove the context layer from all merlin models blocks).  So having the weight tying as a boolean flag won't work as the prediction block doesn't have access to the embeddings table (during the build or the call method). 

we can maybe instead prepare an example notebook that illustrates the common use case (with item-id) ?

Reply:
I see @sararb . Your point makes sense. I agree we should have and example and/or doc showing how to use weight-tying